### DISCUSSION TITLE
Troubleshooting Crazyflie Firmware Customization: M2 LED Remains On

### DISCUSSION THREAD

**Discussion by Vail16768** [bitcraze/discussions#960] [Q&A] [answered]
After successfully flashing the original firmware onto the Crazyflie using the dfu-util command, I decided to make a modification. The only change I made was the addition of an empty function in the code. I then compiled the firmware using the make command and proceeded to upload the newly built firmware onto the Crazyflie. However, upon rebooting the Crazyflie, I noticed that only the M2 LED remained on, while all the other LEDs turned off. What could be causing this behavior? How can I troubleshoot and ensure that my custom firmware runs smoothly on the Crazyflie?

**Comment by knmcguire** [comment, marked as answer]
Ai... that sounds like a bricked crazyflie to me. That is the risk of flashing it with dfu flashing which we warn [about in the documentation](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/). Is there a reason why you don't want to use the over the air (OTA) flashing with crazyradio with [these instructions](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#flashing)? These will ensure that the bootloader will not get broken in the future.

In your case... first go through these steps and see if you are able to snap the Crazyflie out of it: https://www.bitcraze.io/support/troubleshooting/#crazyflie-is-bricked

If that doesn't work, then the bootloader is broken and you'll need to reflash that. That is where these instructions will help out with: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/#bootloader-recovery

This uses dfu as well but that is because this is supposed to only be done once (and it can't be done with the over air flashing).

**Reply by knmcguire** [reply]
@Vail16768 which part of the instructions ended up working for you?

**Reply by Vail16768** [reply]
it was necessary to recover the bootloader.
